cql3/values.hh

```diff
@@ -262,6 +262,18 @@ public:
             }
         }, std::move(_data));
     }
+    managed_bytes_opt to_managed_bytes_opt() && {
+        return std::visit(overloaded_functor{
+            [](bytes&& bytes_val) { return managed_bytes_opt(bytes_val); },
+            [](managed_bytes&& managed_bytes_val) { return managed_bytes_opt(std::move(managed_bytes_val)); },
+            [](null_value&&) -> managed_bytes_opt {
+                return std::nullopt;
+            },
+            [](unset_value&&) -> managed_bytes_opt {
+                return std::nullopt;
```

Comment:
This is suspicious. Maybe we should throw here.

Reply:
I wanted to have behaviour consistent with [`raw_value_view::to_managed_bytes_opt`](https://github.com/scylladb/scylla/blob/dd9d6c081ee48e670eda70e14a765442db62c0ec/cql3/values.hh#L158)

Making it throw would probably be more correct, but I'm worried that some existing code might break.